### DISCUSSION TITLE
Character blacklist/whitelist supported?

### DISCUSSION THREAD

**Discussion by Xargonus** [mindee/doctr#888] [Q&A]
Hello, is there currently a way to blacklist or whitelist characters used by the text recognition model?

**Comment by felixdittrich92** [comment]
Hi @Xargonus 👋 ,
My apologies for the late reply,
currently not but feel free to open an issue for this 👍

**Reply by SlappyAUS** [reply]
This feature is particularly useful in handwriting and form capture. For instance in numeric fields a handwritten "4" outputs as "A". Another issue is bad scans, where paper artefacts can get detected as punctuation.

**Reply by frgfm** [reply]
Hi @SlappyAUS :wave: 

I agree but there are two wider topics at hand here:
- the task definition : whether text recognition can use the hypothesis of having no blanks
- handwriting vs typed : handwriting has a difficulty higher than just variability in character style, the spacing and alignments are not fixed. This is a challenge to match with typed ones.

As of right now, we don't support handwriting yet, but we'll have to circle back to this once we do. I suggest checking docTR again once we add handwriting support :)

**Reply by frgfm** [reply]
Also black/white listing is not exactly possible, there are two fundamental options:
whether the blank is a recognized character or not.

so blacklisting is only possible if the model has blank as a recognized character :man_shrugging:
